### DISCUSSION TITLE
Welcome to k8s-bootstrapper Discussions!

### DISCUSSION THREAD

**Discussion by diabhey** [hivenetes/k8s-bootstrapper#1] [Q&A]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jimtyrro** [comment]
Hi,

I'm trying to follow your tutorial but encountered the following errors when trying to apply my changes with 

`terraform apply -var-file="variables.tfvars.example" --auto-approve`

Error message:

`digitalocean_container_registry.registry: Creating...`
`digitalocean_kubernetes_cluster.cluster: Creating...`

`Error: Error creating Kubernetes cluster: POST https://api.digitalocean.com/v2/kubernetes/clusters: 422 (request "9ea95728-d4cd-4d4a-ab69-19ae818aeb73") validation error: cluster_spec.invalid name`

   `with digitalocean_kubernetes_cluster.cluster,
   on doks.tf line 22, in resource "digitalocean_kubernetes_cluster" "cluster":
   22: resource "digitalocean_kubernetes_cluster" "cluster" {`


`Error: Error creating container registry: POST https://api.digitalocean.com/v2/registry: 422 (request "0ee64a5f-db31-47ee-a972-794e225e17ba") user already has a registry`

   `with digitalocean_container_registry.registry,
   on doks.tf line 43, in resource "digitalocean_container_registry" "registry":
   43: resource "digitalocean_container_registry" "registry" {`

Can you help me to troubleshoot this issue?

Thank you.

**Comment by jimtyrro** [comment]
Never mind, the cluster name had _ character which apparently is not allowed, updated to - and now everything is ok.

**Comment by jimtyrro** [comment]
Another issue. Cluster creation completed successfully according to the message:

`digitalocean_kubernetes_cluster.cluster: Creation complete after 8m24s [id=9be3d016-a1bb-4974-960c-fe91f7fb387e]`

But when I run this command: `doctl kubernetes cluster kubeconfig save 9be3d016-a1bb-4974-960c-fe91f7fb387e`

I shows the following error:

`Error: GET https://api.digitalocean.com/v2/kubernetes/clusters/9be3d016-a1bb-4974-960c-fe91f7fb387e/kubeconfig: 404 (request "80438a43-0169-4e26-bc31-02566fdec8a6") cluster not found`

Any ideas what might have gone wrong?

**Reply by diabhey** [reply]
@jimtyrro : I am pleased to hear that you are trying the bootstrapper. Can you confirm that you have the proper access? I'm just wondering if you have the authentication for doctl to access your account. if not, can you try this https://docs.digitalocean.com/reference/doctl/how-to/install/

**Reply by jimtyrro** [reply]
Thank you for replying. Yes I do have access and can see my account with doctl.

**Reply by jimtyrro** [reply]
But neither doctl nor the DO UI shows any cluster created.

**Reply by diabhey** [reply]
@jimtyrro: It looks like the cluster creation was not successful. Can you try it again? Also, would you mind sharing your "variables.tfvars.example" file to ensure you have configured everything correctly?

**Reply by jimtyrro** [reply]
@diabhey : See "attached variables.tfvars.example" file.

[variables.tfvars.example.txt](https://github.com/hivenetes/k8s-bootstrapper/files/10320474/variables.tfvars.example.txt)

When I'm trying to re-run the cluster creation I'm getting the following:

`terraform apply -var-file="variables.tfvars.example" --auto-approve`

`digitalocean_kubernetes_cluster.cluster: Refreshing state... [id=9be3d016-a1bb-4974-960c-fe91f7fb387e]`

`Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:`
  `+ create`

`Terraform will perform the following actions:`

 ` # digitalocean_container_registry.registry will be created`
  `+ resource "digitalocean_container_registry" "registry" {
      + created_at             = (known after apply)
      + endpoint               = (known after apply)
      + id                     = (known after apply)
      + name                   = "jimtyrrocr2"
      + region                 = (known after apply)
      + server_url             = (known after apply)
      + storage_usage_bytes    = (known after apply)
      + subscription_tier_slug = "basic"
    }`

`Plan: 1 to add, 0 to change, 0 to destroy.
digitalocean_container_registry.registry: Creating...`

`Error: Error creating container registry: POST https://api.digitalocean.com/v2/registry: 422 (request "a3bc62ba-7aa9-4aba-9885-a02637ecd6d9") user already has a registry`

` with digitalocean_container_registry.registry,
   on doks.tf line 43, in resource "digitalocean_container_registry" "registry":
   43: resource "digitalocean_container_registry" "registry" {`

**Reply by diabhey** [reply]
@jimtyrro : Thanks for sharing. I am working on some improvements in the terraform module. I will get back to you as soon as I have cleaned up some stuff! Alternatively, I suggest you check out [this](https://github.com/hivenetes/k8s-bootstrapper#alternatively-set-up-infrastructure-on-digitalocean-using-doctl) so that you can continue using the bootstrapper.

**Reply by diabhey** [reply]
@jimtyrro : There have been some improvements to the terraform module. Checkout the README.md for the updated instructions. If you run into issues, let me know. Thanks!
